src/interactions/slashCommands/mixins/CommandOptions.ts

```diff
@@ -70,7 +70,16 @@ export class SharedSlashCommandOptions<ShouldOmitSubcommandFunctions = true> {
 	 * @param input A function that returns an option builder, or an already built builder
 	 */
 	public addStringOption(
-		input: SlashCommandStringOption | ((builder: SlashCommandStringOption) => SlashCommandStringOption),
+		input:
+			| SlashCommandStringOption
+			| Omit<SlashCommandStringOption, 'setAutocomplete'>
+			| Omit<SlashCommandStringOption, 'addChoice' | 'addChoices'>
```

Comment:
Just as a general question, are these needed to be added?

Reply:
Yeah, otherwise ts will throw if you end your builder with a method that returns one of those.